dependencies.yaml

```
@@ -263,7 +265,7 @@ dependencies:
       - output_types: [conda, requirements]
         packages:
           # Allow runtime version to float up to minor version
-          - libarrow==12.*
+          - libarrow==12.0.1.*=hb6645e0_10_cpu
```

Maybe `12.*` was too flexible

Also wondering if we should split this into Conda & Pip separately and then use `=`s with Conda & `==`s with Pip. It might be trying to consolidate this is loosing some amount of specificity that either tool has that would help with the problem

```suggestion
          - libarrow==12.0.1.*
```

---

I'm not even sure why this is listed for pip, there is no libarrow package for that.